src/System.Security.Cryptography.Pkcs/src/Microsoft/Win32/SafeHandles/SafeProvOrNCryptKeyHandleUwp.cs

```diff
@@ -16,6 +16,19 @@ namespace Microsoft.Win32.SafeHandles
     // 
     internal sealed class SafeProvOrNCryptKeyHandleUwp : SafeProvOrNCryptKeyHandle
     {
+        internal SafeProvOrNCryptKeyHandleUwp(IntPtr handle, SafeHandle parentHandle)
+            : this(handle, true, true)
+        {
+            Debug.Assert(parentHandle != null && !parentHandle.IsClosed && !parentHandle.IsInvalid);
+
+            // If the provided handle value wasn't valid we won't call dispose, so we shouldn't be doing this.
+            Debug.Assert(!IsInvalid);
```

Comment:
I noticed similar code allows Invalid but releases parent handle in such cases (SafeNCryptHandle ctor)